docs/modules/record-encryption/con-lost-kek.adoc

```diff
@@ -0,0 +1,94 @@
+There will be messages like this:
+
+[source]
+----
+Failed to decrypt record in topic-partition <topic>-<partition> owing to key not found condition.
+This will be reported to the client as a RESOURCE_NOT_FOUND(91).
+Client may see a message like 'Unexpected error code 91 while fetching at offset' (java) or or 'Request illegally referred to resource that does not exist' (librdkafka).
+Cause message: key 'd691a642-d8b4-4445-b668-d390df7000bb' is not found (AWS error: ErrorResponse{type='KMSInvalidStateException', message='arn:aws:kms:us-east-1:000000000000:key/d691a642-d8b4-4445-b668-d390df7000bb is pending deletion.'}).
+Raise log level to DEBUG to see the stack.
+----
+
+In this situation, there are three recovery strategies that may be used.
+These are discussed, in order of preference, in the next sections.
+
+== Unschedule key's deletion
+
+Some KMS providers don't actually immediately delete encryption keys.
+Instead, a key is _scheduled_ for deletion at some time in the future.
+When the key is in this state, the key behaves as if it were gone already.
+
+This gives you a grace period where you can change your mind.
+
+Use the Console of the KMS to determine if the missing key is scheduled for deletion.
+If this is the case, use the features of the KMS to undelete the key.
+
+Refer to documentation of the KMS for more details.
+
+After the key has been undeleted, restart the proxy instances.
+Consuming applications should resume as normal.
+
+After the KEK has been recovered from backup and restored to the KMS, restart the proxy instances.
+Applications should begin consuming again as normal.
+
+== Recover the key from backup
+
+If you have a backup of the keys in your KMS, you will be able to recover the key from the backup.
+
+The precise details of how to do this will depend on which KMS you use.
+Refer to documentation of the KMS for more details.
+
+After the KEK has been recovered from backup and restored to the KMS, restart the proxy instances.
+Applications should begin consuming again as normal.
+
+NOTE: When restoring the key to your KMS, it is important to restore the key material and its associated metadata such as its
+identifiers. The Record Encryption filter stores references to the key identifier in the cipher text record. Restoring the key material alone is not sufficient.
+
+== Delete the records encrypted by the lost KEK or advance the consumer groups past them
+
+The final two options are:
+
+* delete the records encrypted with the lost KEK from Kafka, or
+* advance the consumer group offsets used by your applications so that they skip the records that can't be decrypted.
+
+The first challenge is to identify the offsets in the topics after which use of the deleted KEK has ceased.
+If you scale the proxy to more than one replica, the replicas won't necessarily response to a key rotation at the same moment.
```

Comment:
The term replica will make sense for someone using the filter in kube, but on bare metal I don't think replica is typically used to refer to a multi-instance deployment.